code/obj/item/organ_holder.dm

```diff
@@ -1392,3 +1392,148 @@
 		if (istype(M))
 			M.toggle()
 			src.is_on = M.on
+		if(is_on)
+			src.icon_state = initial(src.icon_state)
+		else
+			src.icon_state = "[initial(src.icon_state)]_cd"
+
+
+/datum/targetable/organAbility/kidneypurge
+	name = "Kidney Purge"
+	desc = "Dangerously overclock your cyberkidneys to rapidly purge chemicals from your blood."
+	icon_state = "cyberkidney"
+	targeted = 0
+	cooldown = 40 SECONDS
+	var/power = 6
+
+	cast(atom/target)
+		if (..())
+			return 1
+
+		if(length(linked_organ))
+			for(var/obj/item/organ/O in linked_organ)
+				O.take_damage(15, 15) //safe-ish
+		else
+			linked_organ.take_damage(30, 30) //not safe
+		boutput(holder.owner, "<span class='notice'>You overclock your cyberkidney[islist(linked_organ) ? "s" : ""] to rapidly purge chemicals from your body.")
+		APPLY_MOB_PROPERTY(holder.owner, PROP_CHEM_PURGE, src, power)
+		holder.owner.urine += power // -.-
+		SPAWN_DBG(15 SECONDS)
+			REMOVE_MOB_PROPERTY(holder.owner, PROP_CHEM_PURGE, src)
```

Comment:
This is clever if I'm understanding it right. It's a variable that is applied to the amount of reagents removed each tick. Same with PROP_METABOLIC_RATE I think. Much clearer than my old hardcoded system for kidneys and livers. I'm not familiar with mob properties, I assume that these are allow for variable changes when applying one right.